### ISSUE TITLE
Support Option<Vec<T>> where T is primitive type

### ISSUE THREAD

**Issue by NiwakaDev** [chinedufn/swift-bridge#146]
The following code snippet doesn't seem to work right now:

```rust
#[swift_bridge::bridge]
mod ffi {
    extern "Rust" {
        fn get_option_vector() -> Option<Vec<i32>>;
    }
}
```

If so, can I implement this?

**Comment by chinedufn** [comment]
Yeah, go for it!! Let me know if you have any questions.

**Comment by NiwakaDev** [comment]
Hi, @chinedufn 

I have a quetion related to https://github.com/chinedufn/swift-bridge/blob/master/crates/swift-bridge-ir/src/codegen/codegen_tests/option_codegen_tests.rs#L297-L313.

Is `null pointer` (such as `NULL` in C language) represented as `Optional` in Swift?

According to [Using Imported C Functions in Swift](https://developer.apple.com/documentation/swift/using-imported-c-functions-in-swift), `void *` seem to be represented as `UnsafeMutableRawPointer` in Swift.

**Comment by chinedufn** [comment]
Yeah it's represented as an `UnsafeMutableRawPointer`.

But here when we return `nil` it is getting cast to Swift's `Optional.none`. We could've also returned `.none`.

https://github.com/chinedufn/swift-bridge/blob/4eb92fbb7fbac80ebee47e4a40e2b9d3458611bb/crates/swift-bridge-ir/src/codegen/codegen_tests/option_codegen_tests.rs#L300-L302

Does that answer your question?

**Comment by chinedufn** [comment]
Sorry I mispoke.
`val` is an `Optional<UnsafeMutableRawPointer>`. This is why we need to unwrap it with the `!` in `return SomeType(ptr: val!)`.
So, comparing to nil works and all of our `Optional<T>` integration tests pass https://github.com/chinedufn/swift-bridge/blob/master/SwiftRustIntegrationTestRunner/SwiftRustIntegrationTestRunnerTests/OptionTests.swift#L11

**Comment by chinedufn** [comment]
Closed via #147
